plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractTrinoCatalog.java

```diff
@@ -180,6 +189,41 @@ protected Transaction newCreateTableTransaction(
         return createTableTransaction(schemaTableName.toString(), ops, metadata);
     }
 
+    private boolean namespaceExists(ConnectorSession session, String namespace)
+    {
+        // TODO Improve this when we are improving ConnectorMetadata#schemaExists
+        return listNamespaces(session).contains(namespace);
+    }
+
+    protected abstract boolean tableExists(SchemaTableName schemaTableName);
+
+    protected abstract Optional<String> getOwner(ConnectorSession session);
+
+    @Override
+    public void registerTable(ConnectorSession session, SchemaTableName schemaTableName, String tableLocation, Optional<String> metadataFileName)
```

Comment:
Personal preference, but I'd rather have figured out what the most recent metadata file is before we get here

```suggestion
    public void registerTable(ConnectorSession session, SchemaTableName schemaTableName, String tableLocation, String metadataFileName)
```

Reply:
Moved the file check inside `RegisterTableProcedure` class.